### PR TITLE
Adds `typing_extensions.OrderedDict` alias, refs #11528

### LINKED ISSUE #11528: Variable `typing_extensions.OrderedDict` is not valid as a type
```py
from typing_extensions import OrderedDict

a: OrderedDict[str, str] = {}  # error: Variable "typing_extensions.OrderedDict" is not valid as a type

class A(OrderedDict[str, str]):  #  error: Invalid base class "OrderedDict"
    ...
```
mypy 0.920+dev.6b9cc4774f7dbd556a655071800a8e17e368282b

### PR DESCRIPTION
**Pull request by sobolevn**
I went for `pythoneval` test suite, because all others were very hard to debug / fix.
All of them also required `collections.pyi` hacks, so it was not worth it.

Closes #11528

### CODE REVIEW

**Comment by sobolevn** [comment]
```
 =================================== FAILURES ===================================
__________________________ SamplesSuite.test_samples ___________________________
[gw0] linux -- Python 3.10.0 /home/runner/work/mypy/mypy/.tox/py/bin/python
Sample check failed
----------------------------- Captured stdout call -----------------------------
mypy/typeshed/stdlib/typing_extensions.pyi:70: error: Cannot assign multiple types to name "OrderedDict" without an explicit "Type[...]" annotation
Found 1 error in 1 file (checked 1 source file)
_______________________ SamplesSuite.test_stdlibsamples ________________________
[gw0] linux -- Python 3.10.0 /home/runner/work/mypy/mypy/.tox/py/bin/python
Sample check failed
----------------------------- Captured stdout call -----------------------------
mypy/typeshed/stdlib/typing_extensions.pyi:70: error: Cannot assign multiple types to name "OrderedDict" without an explicit "Type[...]" annotation
Found 1 error in 1 file (checked 12 source files)
=========================== short test summary info ============================
FAILED mypy/test/testsamples.py::SamplesSuite::test_samples
FAILED mypy/test/testsamples.py::SamplesSuite::test_stdlibsamples
====== 2 failed, 9584 passed, 370 skipped, 7 xfailed in 843.66s (0:14:03) ======
ERROR: InvocationError for command /home/runner/work/mypy/mypy/.tox/py/bin/python -m pytest -n 2 (exited with code 1)
```

How can I fix it? 
I think that I need help here 🙂

**Comment by sobolevn** [comment]
I will ping @JelleZijlstra and @hauntsaninja as both typeshed and mypy experts 🙂

**Comment by sobolevn** [comment]
Ok, I've got a simpliest repro:

```
» mypy mypy/typeshed/stdlib/typing_extensions.pyi
mypy/typeshed/stdlib/typing_extensions.pyi:70: error: Cannot assign multiple types to name "OrderedDict" without an explicit "Type[...]" annotation
Found 1 error in 1 file (checked 1 source file)
```

**Comment by JelleZijlstra** [comment]
Sorry, not sure about this. `OrderedDict` is the *only* `_Alias` in typing-extensions, which is probably why we haven't run into this before. Maybe there's some special casing for `typing.pyi` that wasn't carried over.

**Comment by hauntsaninja** [comment]
Try messing around with SemanticAnalyzer.prepare_file

**Comment by sobolevn** [comment]
Yes, I should better prepare `typing_extensions` namespace. I need to remove all `AssignmentStmt` where we define aliases. https://github.com/python/mypy/blob/f316cc2d4b510b79aadb78e0c7fe2814183c93d1/mypy/semanal.py#L340

File: mypy/semanal.py [hunk 1]
```diff
@@ -474,30 +480,53 @@ def add_builtin_aliases(self, tree: MypyFile) -> None:
...
-                    fix_instance_types(target, self.fail, self.note, self.options.python_version)
-                    alias_node = TypeAlias(target, alias,
-                                           line=-1, column=-1,  # there is no context
-                                           no_args=True, normalized=True)
-                    self.add_symbol(name, alias_node, tree)
-            elif self.found_incomplete_ref(tag):
-                # Built-in class target may not ready yet -- defer.
+            self.create_alias(tree, target_name, alias, name)
+
+    def add_typing_extension_aliases(self, tree: MypyFile) -> None:
```
Comment (JelleZijlstra) [inline mypy/semanal.py:485]: This mostly duplicates `add_builtin_aliases`, can you make it take the aliases as a parameter like you did for the prepare method?

Comment (sobolevn) [reply mypy/semanal.py:485]: It is a bit different in this part:

```python
           if type_aliases_source_versions[alias] > self.options.python_version:
                # This alias is not available on this Python version.
                continue
            name = alias.split('.')[-1]
            if name in tree.names and not isinstance(tree.names[name].node, PlaceholderNode):
                continue
```

The thing is: `add_builtins_aliases()` is used to refine `Partial` types as well. But, with the new method - we don't have this problem.

So, I've decided to devide them, there's no clear way to unify this logic in my head 😞

**Comment by sobolevn** [comment]
Thanks everyone! 🎉

**Comment by lmignon** [comment]
Hi,

Is this fix already available into the last release of Mypy? I still have an issue with `typing_extensions.OrderedDict`
```bash
Variable "typing_extensions.OrderedDict" is not valid as a type
```

Regards

**Comment by sobolevn** [comment]
No, this will be available in 0.930 as far as I know.

**Pull request merged**